### PR TITLE
SONAR-6846 support duplicate snapshots with islast=true

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [FeedFileSources.java#L244](https://github.com/SonarSource/sonarqube/blob/1f60bb47e9014ebdd53173e8a553f13015ffc8ce/sonar-db/src/main/java/org/sonar/db/version/v50/FeedFileSources.java#L244): Make this anonymous inner class a lambda [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1604)
1. MAJOR [ws](https://github.com/SonarSource/sonarqube/blob/1f60bb47e9014ebdd53173e8a553f13015ffc8ce/server/sonar-server/src/main/java/org/sonar/server/updatecenter/ws): Add a 'package-info.java' file to document the 'ws' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by simonbrandhof** [comment]
The JIRA ID is not correct. It should be SONAR-7997.

File: sonar-db/src/test/java/org/sonar/db/version/v50/FeedFileSourcesTest.java [hunk 1]
```diff
@@ -89,6 +89,43 @@ public void migrate_ignores_duplicate_multiple_rows_in_snapshot_source_for_the_s
...
+      <snapshots id="6" project_id="3" parent_snapshot_id="4" root_project_id="1" root_snapshot_id="2"
+             status="P" islast="[true]" purge_status="1"
+             period1_mode="days1" period1_param="30" period1_date="2011-09-24"
+             period2_mode="days2" period2_param="31" period2_date="2011-09-25"
+             period3_mode="days3" period3_param="32" period3_date="2011-09-26"
+             period4_mode="days4" period4_param="33" period4_date="2011-09-27"
+             period5_mode="days5" period5_param="34" period5_date="2011-09-28"
+             depth="1" scope="FIL" qualifier="FIL" created_at="2008-12-02" build_date="2011-09-29"
+             version="2.1-SNAPSHOT" path="1.2."/>
+     */
```
Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/version/v50/FeedFileSourcesTest.java]: I suppose that this duplication of before.xml is on purpose to help readability ?

**Comment by simonbrandhof** [comment]
LGTM

File: sonar-db/src/test/java/org/sonar/db/version/v50/FeedFileSourcesTest.java (hunk 1 above)
Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/version/v50/FeedFileSourcesTest.java]: nope, I copied/pasted the XML to write it's java replacement and forgot to remove it

**Pull request merged**